home/service/details.py

```diff
@@ -56,40 +57,94 @@ def _get_context(self):
         return context
 
 
-class DatasetDetailsService(GenericService):
+class DatabaseDetailsService(GenericService):
     def __init__(self, urn: str):
-        super().__init__()
-
+        self.urn = urn
         self.client = self._get_catalogue_client()
 
-        self.table_metadata = self.client.get_table_details(urn)
-
-        # Use a search query to obtain the remaining metadata about the table.
-        # This could be merged into the above query.
         filter_value = [MultiSelectFilter("urn", [urn])]
         search_results = self.client.search(query="", page=None, filters=filter_value)
 
         if not search_results.page_results:
             raise ObjectDoesNotExist(urn)
 
         self.result = search_results.page_results[0]
-        data_products = self.result.metadata.get("data_products", ())
-        if data_products:
-            # Pick the first data product to use as the parent in the breadcrumb.
-            # If the dataset belongs to multiple data products, this may diverge
+        self.entities_in_database = self._get_database_entities()
+        self.context = self._get_context()
+
+    def _get_database_entities(self):
+        # we might want to implement pagination for data product children
+        # details at some point
+        database_search = self.client.list_database_tables(
+            urn=self.urn, count=500
+        ).page_results
+
+        entities_in_database = []
+        for result in database_search:
+            entities_in_database.append(
+                {
+                    "name": result.name,
+                    "urn": result.id,
+                    "description": result.description,
+                    "type": "TABLE",
+                }
+            )
+
+        entities_in_database = sorted(entities_in_database, key=lambda d: d["name"])
+
+        return entities_in_database
+
+    def _get_context(self):
+        context = {
+            "result": self.result,
+            "result_type": (
+                "Database"
```

Comment:
isn't the result_type always database here?

Reply:
Yeah i think it's not needed and was accommodating dodgy test inputs rather than actually being of any use.

Reply:
have committed fix